crmsh/sh.py

```diff
@@ -44,15 +46,26 @@ def __init__(self, msg, cmd):
 class AuthorizationError(Error):
     def __init__(self, cmd: str, host: typing.Optional[str], user: str, msg: str):
         super().__init__(
-            'Failed to run command on {optional_user}{host}: {msg}: {cmd}'.format(
+            'Failed to run command {cmd} on {optional_user}{host}: {msg} {diagnose}'.format(
                 optional_user=f'{user}@' if user is not None else '',
-                host=host, msg=msg, cmd=cmd
+                host=host, msg=msg, cmd=cmd,
+                diagnose=self.diagnose(),
             ),
             cmd
         )
         self.host = host
         self.user = user
 
+    @staticmethod
+    def diagnose() -> str:
+        if user_of_host.instance().use_ssh_agent():
+            with StringIO() as buf:
+                if 'SSH_AUTH_SOCK' not in os.environ:
+                    buf.write('Environment variable SSH_AUTH_SOCK does not exist.')
+                    if 'SUDO_USER' in os.environ:
+                        buf.write(' Please make sure environment variables are preserved across sudo calls.')
```

Comment:
```suggestion
                        buf.write(' Probably could consider, eg. sudo --preserve-env=SSH_AUTH_SOCK .')
```

Reply:
Lacking `--preserve-env=SSH_AUTH_SOCK` is only one of the possible causes of `SSH_AUTH_SOCK` not found in environment variables. How about ` Please check whether an ssh-agent is available and consider using "sudo --preserve-env=SSH_AUTH_SOCK"`

Reply:
`Please check whether ssh-agent is available and consider using "sudo --preserve-env=SSH_AUTH_SOCK"`